test/shared_utils/reference_hashing.py

```diff
@@ -187,7 +191,21 @@ def get_hash_value(
     return sha256(all_bytes).hexdigest()
 
 
-def get_dimensions_bytes(variable_dimensions: tuple[str]) -> bytes:
+def get_group_dimensions_bytes(group_dimensions: tuple[str]) -> bytes:
+    """Convert xarray.DataTree.dims or xarray.Dataset.dims to bytes.
+
+    This differs from `get_variable_dimensions_bytes` by imposing an ordering
+    on the tuple of dimension. With variables, the intrinsic order of the
+    dimensions in important, as they define the ordering of array axes. That
+    same ordering is not important within a group for the purposes of
+    comparison, as there is no effective functional difference observed by
+    end users if dimensions are stored in a different order in an output file.
+
+    """
+    return str(sorted(group_dimensions)).encode('utf-8')
```

Comment:
I'd love someone to just give me a head-nod here that what I'm asserting isn't missing something subtle. My take is that dimensions on a group are just being lumped in there for storage purposes, and for our testing we don't care about the ordering with which that is done. (Distinct from variable dimension ordering, which is definitely important!)

Reply:
That passes the sniff test to me.  I had thoughts of maybe this should have been optional. But I think I've come around to it's fine.  I'm still a little confused why the unsorted items failed. They seem like they should have been preserved by nature of dicts, but I haven't looked into it.